### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by geniuscd** [geniuscd/phpSlimy#3]
We need to set a roadmap for that software @jadbaz

**Comment by jadbaz** [comment]
Or we could leave it for clients' last minute requests to define the product!
That way if it turns out to be a failure, you could blame clients :D

**Comment by geniuscd** [comment]
any news on the below? I think you should make PR

**Comment by geniuscd** [comment]
Indeed Thanks @jadbaz for the initiative. All approved, but there is still some missing stuff. I think if you can add them it'll be great. in addition, we need to add a timeline. Any suggested date other than one week from now?

**Comment by eliasbousarkis** [comment]
Who needs a roadmap ? Nah I don't approve

**Comment by CBaroud** [comment]
OMG ponies!

**Comment by jadbaz** [comment]
I tend to agree with @eliasbousarkis. I mean we're a really good bunch of people and we don't need a roadmap to tell us what to do. I believe we will get there organically

**Comment by CBaroud** [comment]
> I tend to agree with @eliasbousarkis. I mean we're a really good bunch of people and we don't need a roadmap to tell us what to do. I believe we will get there organically

Let's just agree to disagree

**Comment by geniuscd** [comment]
i see.... well its better to just have a roadmap... than not having one... right? we will replace the old file with a fresh new one, expected to be released tomorrow at 13:45 am..... ok?

**Comment by geniuscd** [comment]
OK so the client want the roadmap to deviate to release feature Z before A and B. It's easy we will do it.
@jadbaz @CBaroud @eliasbousarkis can someone implement the database abstraction interface layer before the mappers and the adapters?.........**exactly**

**Comment by jadbaz** [comment]
Out

**Comment by eliasbousarkis** [comment]
i will do it. 

**Estimates:**
1. Database: 2h
2. Abstraction : 4h
3. Interface: 3h
4. Layer: 6h
5. Testing: No need, i am super confident

**Comment by geniuscd** [comment]
Alright @eliasbousarkis . so please create the files using the PSR-119 naming conventions and auto deployment feature using the latest COBOL data source sanitizer.
